File: src/forgesight/api/schemas/vision.py
"""Request/response schemas for the vision/CV endpoints."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Optional

from pydantic import ConfigDict
from sqlmodel import SQLModel


class CvFindingResponse(SQLModel):
    cv_finding_id: uuid.UUID
    image_id: uuid.UUID
    board_id: str
    defect_type: str
    component_designator: Optional[str] = None
    confidence: float
    bounding_box: list[int]
    raw_image_reference: str
    model_name: str
    model_version: str
    inference_timestamp: datetime
    dataset_used_for_training: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class InspectionImageResponse(SQLModel):
    image_id: uuid.UUID
    board_id: str
    image_reference: str
    station_id: Optional[str] = None
    captured_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InspectionResultResponse(SQLModel):
    inspection_image: InspectionImageResponse
    cv_findings: list[CvFindingResponse]
    findings_count: int

File: frontend/forgesight-web/src/types/vision.ts
export interface CvFindingResponse {
  cv_finding_id: string;
  image_id: string;
  board_id: string;
  defect_type: string;
  component_designator: string | null;
  confidence: number;
  bounding_box: [number, number, number, number]; // [x, y, w, h]
  raw_image_reference: string;
  model_name: string;
  model_version: string;
  inference_timestamp: string;
  dataset_used_for_training: string | null;
}

export interface InspectionImageResponse {
  image_id: string;
  board_id: string;
  image_reference: string;
  station_id: string | null;
  captured_at: string;
}

export interface InspectionResultResponse {
  inspection_image: InspectionImageResponse;
  cv_findings: CvFindingResponse[];
  findings_count: number;
}